app/static/js/form.js

```diff
@@ -34,4 +34,56 @@ $(document).ready(
           $("#config-input").css("display","block");
         });
 });
+
+function validate() {
+  $("[id=error]").hide();
+
+  var csv = $("textarea[name='csv']").val();
+  var csvFile = $('input[type=file][name="file"]').val();
+
+  if (csv === "" && csvFile === "") {
+    $(".no-data-error").show();
+    return false;
+  }
+
+  if (csv !== undefined && csv !== null && csv !== "") {
+    var csvLines = csv.split("\n");
+    var countLines = 0;
+    csvLines.forEach(function(csvLine) {
+      var line = csvLine.split(",");
+      if (line.length === 4) {
+        countLines += 1;
+      }
+    });
+    if (countLines !== csvLines.length) {
```

Comment:
Can you please explain why we are counting individual lines. I think `csvLines` will always be equal to `countLines`

Reply:
@jajodiaraghav it checks whether all the lines are of the format _,_,_,_,
This same logic is implemented in the server side counterpart, I have just ported it to JS.

Reply:
Okay. Right! Thanks 

Looks good!